Inconsitent legislation patterns

lindadelacombaz opened the pull request:
Dear Mirja and Mauro,

Here is the PR referring to the [legal designation patterns ticket #18](https://github.com/orgs/2DegreesInvesting/projects/13/views/10?pane=issue&itemId=10875625).

I have used the stringr::replace_all function that recognizes patterns like s.l. and s l and replace it to sl (but **not** a company that it called peter**s l**amborghini --> will not be peter**sl** amborghini).

lindadelacombaz committed:
replaced spanish legislation patterns

lindadelacombaz committed:
fixed styling mistake

lindadelacombaz committed:
removed |

maurolepore commented:
Thanks @lindadelacombaz, 

It's looking good but it'd be great to add some tests every time we touch an exported function. To do that it would be great if you could first review these two PRs:

* #17 
* #15 

Also, this seems pretty straight forward and my review will include some tests, so I think my review should be enough so Mirja can focus on other things (unless you @MirijaHa do want to review too).

lindadelacombaz committed:
merging main with branch

lindadelacombaz committed:
removed stringr - changed strategy

lindadelacombaz committed:
incorporated spanish legislations into from_to table

lindadelacombaz committed:
added the test for spanish legislations

lindadelacombaz committed:
error ls into sl

lindadelacombaz commented:
Hi again @maurolepore , thank you for the great comments.

- I have merged the branch with your PRs modifications (namely having an individual R script for to_alias.R and a test file)

- I have also changed the strategy to remove the inconsistent Spanish legislations by just adding "s.l." and "s l" into the from_to_alias table.

- I have added a test subsequently to control the changes.

I think it this approach is better than my last one, because :

1. It does not need an additional package (stringr) for the bank to install
2. It is consistent with the logic of how the function is made, namely list all kinds of legislations patterns / designations and formalize them.

maurolepore committed:
Add test cases and reword test title

maurolepore commented on R/to_alias.R:
```diff
@@ -245,6 +245,8 @@ from_name_to_alias <- function() {
     "shipping",             "shp",
     "sirketi",             "sti",
+    "s.l.",                "sl",
+    "s l",                 "sl",
```
Revert

maurolepore commented on tests/testthat/test-to_alias.R:
```diff
@@ -183,6 +193,8 @@ pacta_data_name_reductions <- tibble::tribble(
   "san tic ltd sti",    "santicltdsti",
   "san tic anonim sti", "santicanonimsti",
+  "s.l.",                    "sl",
+  "s l",                     "sl",
```
Revert

maurolepore reviewed (changes requested):
Linda, I just realized `to_alias()` already behaved the way we want. I would leave the tests to show we meet the requirement but please delete the code.

This highlights the importance not only of tests but also of writing the tests before the code (TDD: Test Driven Development). TDD always starts with a failing test. If you write the test and it doesn't fail, it's either because the test is wrong or because the code already works as you expect.

``` r
to_alias_old <- r2dii.match:::to_alias
to_alias_old("abc s.l.")
#> [1] "abcsl"
to_alias_old("abc s l")
#> [1] "abcsl"
to_alias_old("abc sl")
#> [1] "abcsl"
# Same
to_alias_new <- tilt.company.match::to_alias
to_alias_old("abc s.l.")
#> [1] "abcsl"
to_alias_old("abc s l")
#> [1] "abcsl"
to_alias_old("abc sl")
#> [1] "abcsl"
```

<sup>Created on 2023-02-07 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

lindadelacombaz committed:
reverted changes

lindadelacombaz committed:
pushing references problem
Merge branch 'legislation_patterns' of github.com:2DegreesInvesting/tilt.company.match into legislation_patterns

lindadelacombaz committed:
fix. merge mistake

lindadelacombaz committed:
removed ls

lindadelacombaz commented:
Dear @maurolepore, thank you for this great catch and comment !!
Will apply the TDD in the future for sure ! :)

maurolepore reviewed (approved):
LGTM :+1:

maurolepore commented:
@lindadelacombaz rather than forcing yourself to always test first I suggest you at lease consider it as one of your options. See [When should you write tests?](https://mastering-shiny.org/scaling-testing.html#when-should-you-write-tests) -- by Hadley Wickham.
